src/rust/ide/ast/impl/src/known.rs

```diff
@@ -58,6 +58,16 @@ impl<T> KnownAst<T> {
           E                      : Debug, {
         self.deref()
     }
+
+    /// Returns AST with Shape modified with the provided method.
+    pub fn map_shape<E>(&self, f:impl FnOnce(&mut T)) -> KnownAst<T>
+    where for<'t> &'t Shape<Ast> : TryInto<&'t T,Error=E>,
+          T                      : Clone + Into<Shape<Ast>>,
+          E                      : Debug {
+        let mut shape = self.shape().clone();
+        f(&mut shape);
+        Self::new(shape, self.id())
+    }
```

Comment:
Well, after our discussion about `map` methods, this should be called "map in place", or `f` could return any shape type U (and map_shape would return `KnownAst<U>`)

That's only a suggestion